src/unistd.rs

```diff
@@ -3151,6 +3151,139 @@ impl User {
             unsafe { libc::getpwnam_r(name.as_ptr(), pwd, cbuf, cap, res) }
         })
     }
+
+    /// Construct an iterator over all users on the system.
+    ///
+    /// Call [`setpwent(3)`](https://pubs.opengroup.org/onlinepubs/7908799/xsh/setpwent.html),
+    /// [`getpwent(3)`](https://pubs.opengroup.org/onlinepubs/7908799/xsh/getpwent.html),
+    /// [`getpwent_r(3)`](https://man7.org/linux/man-pages/man3/getpwent_r.3.html) and
+    /// [`endpwent(3)`](https://pubs.opengroup.org/onlinepubs/9699919799/functions/endpwent.html)
+    /// under the hood.
+    ///
+    /// When encounters an error, [`AllUsers::next()`] will yield a
+    /// `Some(Err(Errno))`, stop iterating if this happens.
+    ///
+    /// # Safety
+    ///
+    /// This function is marked as `unsafe` cause `setpwent()` and `endpwent()`
+    /// will modify a global state, which will result in a data race if we have
```

Comment:
What kind of data race?  Skipping over some users, or duplicating some, is not considered `unsafe`.  Only something like reading a struct with some entries from one source and some from another would be considered `unsafe`.

Reply:
There are two issues here I think:
1. The pointer returned by `getpwent()` points to static memory, and thus can be overwritten by subsequent calls like `getpwent`, `getpwuid`, and `getpwnam`. So in the implementation using `getpwent()`, when executing `Some(Ok(User::from(unsafe{&*pwd})))` in a thread (say thread A), there happens to be another thread (thread B) calling `getpwent/getpwuid/getpwnam`, then the static memory will be overwritten due to the call in thread B, making thread A get a wrong result. This is the data race.
2. `setpwent/endpwent` will modify a global state, if two instances are running at the same time, the state set by the first instance can be reset by the second one, then the first instance will get duplicate entries. 

Sorry for the unclarity in my documentation, will update it stating both issues.